docs/fundamentals/executor/monitoring-executor.md

```diff
@@ -0,0 +1,168 @@
+
+````{admonition} Using @monitor
+:class: hint
+Adding the custom monitoring on a method is as straightforward as decorating the method with `@monitor` 
+````
+
+```python
+from jina import Executor, monitor
+
+
+class MyExecutor(Executor):
+    @monitor()
+    def my_method(self):
+        ...
+```
+
+This will create a [Prometheus summary](https://prometheus.io/docs/concepts/metric_types/#summary)
+`jina_my_method_inference_seconds` which will keep track of the time of execution of `my_method
+
+By default, the name and the documentation of the metric created by `@monitor` are auto-generated based on the name
+of the function. However, you can precise it by yourself by doing :
+
+```python
+@monitor(
+    name='my_custom_metrics_seconds', documentation='This is my custom documentation'
+)
+def method(self):
+    ...
+```
+
+````{admonition} respect Prometheus naming
+:class: caution
+You should respect Prometheus naming [conventions](https://prometheus.io/docs/practices/naming/#metric-names). 
+therefore because `@monitor` creates a [Summary](https://prometheus.io/docs/concepts/metric_types/#summary) under the hood
+your metrics name should finish with `seconds`
+````
+
+## Example
+
+Let's take an example to illustrate custom metrics:
+
+```python
+from jina import requests, Executor
+from docarray import DocumentArray
+
+
+class MyExecutor(Executor):
+    def preprocessing(self, docs: DocumentArray):
+        ...
+
+    def model_inference(self, tensor):
+        ...
+
+    @requests
+    def encode(self, docs: DocumentArray, **kwargs):
+        docs.tensors = self.preprocessing(docs)
+        docs.embedding = self.model_inference(docs.tensors)
+```
+
+This Executor encodes is composed of two sub-functions.
+* `preprocessing` which takes raw bytes from a DocumentArray and put them into a torch tensor. 
+* `model inference` calls the forward function of a deep learning model.
+
+By default, only the encode function will be monitored. 
+
+```{code-block} python
+---
+emphasize-lines: 6, 10
+---
+from jina import requests,monitor,Executor
+from docarray import DocumentArray
+
+class MyExecutor(Executor):
+
+    @monitor()
+    def preprocessing(self, docs: DocumentArray):
+        ...
+
+    @monitor()
+    def model_inference(self, tensor):
+        ...
+
+    @requests
+    def encode(self, docs: DocumentArray, **kwargs):
+        docs.tensors = self.preprocessing(docs)
+        docs.embedding = self.model_inference(docs.tensors)
+```
+
+
+### Defining custom metrics directly with the Prometheus client
+
+Under the hood, the monitoring feature of the Executor is handled by the 
+python [Prometheus-client](https://github.com/prometheus/client_python). The `@monitor` decorator is a convenient tool
+to monitor sub-method of an Executor, but you might need more flexibility and that is why you can access the Prometheus
+client directly from the executor to define every kind of metric supported by Prometheus.
```

Comment:
```suggestion
client directly from the Executor to define every kind of metric supported by Prometheus.
```

Always do a search for `executor`, `flow`, etc in your docs to ensure you don't miss anything like this